ddl/db_change_test.go

```diff
@@ -1991,3 +1992,120 @@ func (s *stateChangeSuite) TestRestrainDropColumnWithIndex() {
 	tk.MustExec("alter table t drop column a;")
 	tk.MustExec("drop table if exists t;")
 }
+
+func TestParallelRenameTable(t *testing.T) {
+	store, d, clean := testkit.CreateMockStoreAndDomain(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("create database test2")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int default 0, b int default 0, key idx((b+1)))")
+	tk1 := testkit.NewTestKit(t, store)
+	tk1.MustExec("use test")
+	tk3 := testkit.NewTestKit(t, store)
+	tk3.MustExec("use test")
+
+	var concurrentDDLQueryPre string
+	var concurrentDDLQuery string
+	firstDDL := true
+
+	var wg sync.WaitGroup
+	var checkErr error
+	d2 := d.DDL()
+	originalCallback := d2.GetHook()
+	defer d2.SetHook(originalCallback)
+	callback := &ddl.TestDDLCallback{Do: d}
+	callback.OnJobRunBeforeExported = func(job *model.Job) {
+		switch job.SchemaState {
+		case model.StateNone:
+			if firstDDL {
+				firstDDL = false
+			} else {
+				return
+			}
+			wg.Add(1)
+			go func() {
+				if concurrentDDLQueryPre != "" {
+					go func() {
+						// We assume that no error, we don't want to test it.
+						tk3.MustExec(concurrentDDLQueryPre)
+					}()
+					time.Sleep(10 * time.Millisecond)
+				}
+				_, err := tk1.Exec(concurrentDDLQuery)
```

Comment:
It seems that the test is not quite sure whether it started from the state of `StateNone`  `RenameTable` or after `RenameTable` was executed.

Reply:
Sleep is good enough for this test case. In the worse case, it won't cause the test to fail.